### ISSUE TITLE
Building pdf on CI

### ISSUE THREAD

**Issue by 153957** [davidfokkema/artist#30]
Today I came across [this repo](https://github.com/tompollard/phd_thesis_markdown/) in which someone  [installs](https://github.com/tompollard/phd_thesis_markdown/blob/master/.travis.yml#L21) a newer version of TeX Live on Travis (this takes only [± 3 minutes](https://travis-ci.org/tompollard/phd_thesis_markdown#L106)). This is the 2012 version of TeX Live.

We could use this to test the compilation of the .tex files, see #14.

**Comment by 153957** [comment]
And if we choose not to do this we should add `sudo: false` to the `.travis.yml` files to enable container based builds.

**Comment by 153957** [comment]
See 'working' example in the ci-tex branch..
This does take several more minutes (9 instead of 2) to run.
And not every demo works, I skipped the demo that required SciPy, disabled the Matplotlib plot, and skipped the Polar axis plots, which use a coordinate system to position the xlabel which is apparently unknown to the TikZ/PGFPlots in TeXLive 2012.

So.. is it worth the hassle? at least you will know if the pdf can compile.
I only found out about the bug fixed with aad966be9ef1c3ed8982a2cb0f2390bb8726556d because I manually compiled the entire demo, the making of the tex files worked fine. However, that was not really an artist bug, but a LaTeX bug.

**Comment by 153957** [comment]
Note; do not merge the ci-tex branch, it is a bit messy due to testing things, and the demo is no longer complete.

**Comment by davidfokkema** [comment]
Do you know of any PPA's with TeXLive 2015 (or 2014?). 2012 is a bit old. If we do this, we should decide which versions of TeXLive to support. Suppose we decide to support the current and previous versions, then we should test for TeXLive 2015 *and* TeXLive 2014. Is that feasible?

**Comment by 153957** [comment]
I do not, I don't think there is any which will work on Travis. So supporting TeXLive on Travis remains unlikely.

**Comment by davidfokkema** [comment]
Pity. Is it possible to upload stuff to Travis? Or ssh into an environment? Maybe we can install TeXLive (both versions) manually and tar.gz the whole tree for unpacking during builds.

**Comment by davidfokkema** [comment]
Is this something we can use? https://github.com/thomasjo/travis-texlive. Not nice, but it works, I guess. Or create a custom package. Lots of work.

**Comment by 153957** [comment]
Yea, that might be difficult.
Travis also has some caching mechanism allowing you to cache some packages that are used for each build, Im not certain about the specifics, but that might be useful if we use our own built versions.

**Comment by 153957** [comment]
See:
https://github.com/HiSPARC/infopakket/pull/43/
https://github.com/HiSPARC/routenet/pull/6/
https://github.com/HiSPARC/logo/pull/4/

**Comment by 153957** [comment]
Or maybe easier using Tectonic: https://malramsay.com/post/compiling_latex_on_travis/

**Comment by davidfokkema** [comment]
Looks interesting!

**Comment by 153957** [comment]
Indeed, it should automatically install the packages required for the document you throw at it, so no worrying about configuring the install and it should keep the installation relatively small.

**Comment by davidfokkema** [comment]
Yes, the only gripe I have so far is that it doesn’t build on lualatex, de official successor of pdflatex. I have a large TikZ figure which both pdflatex and xelatex won’t compile because of memory problems, while lualatex just takes a long time. Oh well...

Verstuurd vanaf mijn iPad

**Comment by 153957** [comment]
One of the 👎 for tectonic is that it does not support the 'shell-escape' option, i.e. it does not support the popular `minted` package: https://github.com/tectonic-typesetting/tectonic/issues/38

This is usually not required for rendering figures generated with artist, but might be for other parts of documents which include artist figures.

**Comment by davidfokkema** [comment]
That will be a problem if we ever decide to build a TikZ-like pdf manual. For now, let's stick to Sphinx then.
